src/Orleans/Providers/ProviderTypeLoader.cs

```diff
@@ -87,6 +87,14 @@ private void ProcessAssemblyLocally(Assembly assembly)
 
         private static void ProcessNewAssembly(object sender, AssemblyLoadEventArgs args)
         {
+#if !NETSTANDARD
+            // If the assembly is loaded for reflection only avoid processing it.
+            if (args.LoadedAssembly.ReflectionOnly)
```

Comment:
The linked issue doesn't look like it was related to the ProviderTypeLoader. What led you to here? 
Also, I think that because this is wired to the `AppDomain.AssemblyLoad` event, it should not be getting called when a reflection-only assembly is loaded (I might be wrong, because there's too much black magic with this context, but I think that's what I remember).